### PR TITLE
Project details

### LINKED ISSUE #159: Project details page main content
Fetch from API(s) and render content

### PR DESCRIPTION
**Pull request by alanmoo**
Closes #159

### CODE REVIEW

File: app/react/components/categories/categories.jsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+
+import React from "react";
+
+export default React.createClass({
+  propTypes: {
+    categories: React.PropTypes.array
```
Comment (gvn) [inline app/react/components/categories/categories.jsx]: This is required, right?

File: app/react/pages/project-details/project-details.jsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+  getInitialState() {
+    return {
+      projectDetails: {}
+    };
+  },
+  componentWillMount() {
+    this.getProjectDetails();
+  },
+  getProjectDetails() {
```
Comment (gvn) [inline app/react/pages/project-details/project-details.jsx]: Can you just add a TODO comment to use the API module once we've landed it?

File: app/react/pages/project-details/project-details.jsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+    xhr.open(`GET`, url);
+    xhr.responseType = `json`;
+
+    xhr.onload = () => {
+      this.setState({projectDetails: xhr.response});
+    };
+
+    xhr.onerror = () => {
+      console.log(`Error fetching projects`);
```
Comment (gvn) [inline app/react/pages/project-details/project-details.jsx]: Use `console.error`

File: app/react/pages/project-details/project-details.jsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+                return <GithubAvatar key={contributor.username} user={contributor} />;
+              }): null }
+            </div>
+          </div>
+          <ProjectCardFeatured project={project} />
+          <div className="col-xs-12 col-md-10 col-md-push-1">
+            <h3>What we're doing</h3>
+            <p>{project.description}</p>
+            <div className="text-xs-center p-t-1 p-b-3">
+              <a href="#" className="btn visit-repo">VISIT REPO</a>
```
Comment (gvn) [inline app/react/pages/project-details/project-details.jsx]: I usually write stuff like this in standard case, and use `text-transform: uppercase`. Not a huge deal though.

File: app/react/pages/project-details/project-details.scss [hunk 5]
```diff
@@ -0,0 +1,12 @@
+#project-details {
+  .github-avatar {
```
Comment (gvn) [inline app/react/pages/project-details/project-details.scss]: Is some of this duplicated at: https://github.com/mozilla/science.mozilla.org/pull/260/files#diff-77639168fc75707c152b83e4b810df2eR1 ?

**Comment by gvn** [comment]
Looking awesome. Just a few nits/tweaks and R+!

**Comment by gvn** [comment]
Actually, now that we have the service module, can you refactor to use it?

File: app/react/pages/project-details/project-details.jsx (hunk 2 above)
Comment (gvn) [inline app/react/pages/project-details/project-details.jsx]: Refactor w. service module...

File: app/react/pages/project-details/project-details.scss (hunk 5 above)
Comment (alanmoo) [inline app/react/pages/project-details/project-details.scss]: Oh weird, I thought I got rid of it from here when I moved it.

File: app/react/pages/projects/projects.jsx [hunk 6]
```diff
@@ -22,21 +24,10 @@ export default React.createClass({
...
-      this.setState({projects: xhr.response});
-    };
-
-    xhr.onerror = () => {
-      console.log(`Error fetching projects`);
-    };
-
-    xhr.send();
+    Service.projects
+      .get()
```
Comment (gvn) [inline app/react/pages/projects/projects.jsx]: Just need to pass an object here w. your query.

**Comment by gvn** [comment]
👍

**Pull request merged**